packages/vuetify/src/composables/loader.tsx

```diff
@@ -0,0 +1,52 @@
+// Components
+import { VProgressLinear } from '@/components/VProgressLinear'
+
+// Utilities
+import { computed } from 'vue'
+import { propsFactory } from '@/util'
+
+// Types
+import type { Slot } from 'vue'
+
+export interface LoaderProps {
+  loading: boolean
+}
+
+// Composables
+export const makeLoaderProps = propsFactory({
+  loading: Boolean,
+}, 'loader')
+
+export function useLoader (props: LoaderProps, name: string) {
+  const loaderClasses = computed(() => ({
+    [`${name}--loading`]: props.loading,
+  }))
+
+  return { loaderClasses }
+}
+
+// Generators
+export function genLoader (
```

Comment:
I thought we were moving away from gen functions. This could be a functional component. Potentially even returned by `useLoader`. For example, that way you don't have to pass 'v-field' twice.

Reply:
I based the process off of variant.ts's genOverlays method. As far as exporting it from the use function, we decided against doing that for variant, but it may be applicable here, would need @KaelWD to chime in.